probe-rs/src/bin/probe-rs/cmd/complete.rs

```diff
@@ -0,0 +1,219 @@
+use std::io::{Cursor, Write};
+
+use anyhow::{anyhow, Context};
+use clap::CommandFactory;
+use clap_complete::{generate, Shell};
+use probe_rs::probe::list::Lister;
+
+use crate::{util::common_options::OperationError, Cli};
+
+const BIN_NAME: &str = "probe-rs";
+
+/// Install and complete autocomplete scripts
+#[derive(clap::Parser)]
+#[clap(verbatim_doc_comment)]
+pub struct Cmd {
+    #[clap(long)]
+    shell: Option<Shell>,
+    #[clap(subcommand)]
+    kind: CompleteKind,
+}
+
+impl Cmd {
+    pub fn run(&self, lister: &Lister) -> Result<(), anyhow::Error> {
+        let shell = Shell::from_env()
+            .or(self.shell)
+            .ok_or_else(|| anyhow!("The current shell could not be determined. Please specify a shell with the --shell argument."))?;
+        if !matches!(shell, Shell::Zsh | Shell::Bash) {
+            anyhow::bail!("Only ZSH and Bash are supported for autocompletions at the moment");
+        }
+
+        match &self.kind {
+            CompleteKind::Install => {
+                let mut command = <Cli as CommandFactory>::command();
+                let name = std::env::args_os().next().unwrap();
+                let name = name.to_str().unwrap().split('/').last().unwrap();
+                command = command.name("probe-rs");
+                let mut script = Cursor::new(Vec::<u8>::new());
+                generate(shell, &mut command, name, &mut script);
+                let mut script = String::from_utf8_lossy(&script.into_inner()).to_string();
+                inject_dynamic_completions(shell, name, &mut script)?;
+
+                match shell {
+                    Shell::Zsh => {
+                        let Some(dir) = directories::UserDirs::new() else {
+                            eprintln!("User home directory could not be located.");
+                            eprintln!("Install script in ~/.zfunc/_{BIN_NAME}");
+                            println!("{script}");
+                            return Ok(());
+                        };
+                        let dir = dir.home_dir();
+                        std::fs::write(dir.join(format!(".zfunc/_{BIN_NAME}")), &script)
+                            .context("Writing the autocompletion script failed.")?;
+                    }
+
+                    Shell::Bash => todo!(),
```

Comment:
I don't think we should land `todo!()`s :) If this is intentional, please replace with `unimplemented!("justification")`